src/collectors/snmpinterfacepoll/snmpinterfacepoll.py

```diff
@@ -91,17 +95,21 @@ def publish_metrics(self, gaugesList, countersList, timestamp):
                     value=metricValue,
                     oldUnit='byte',
                     newUnit=unit)
-                    metric = Metric(
-                            metricPathNewUnit,
-                            self.derivative(
-                                            metricPathNewUnit,
-                                            metricValueNewUnit,
-                                            COUNTER_MAX_64BIT
-                                            ),
-                            raw_value=None,
-                            timestamp=timestamp
-                            )
-                    self.publish_metric(metric)
+                    # Fixes devices that return no value
+                    try:
+                        metric = Metric(
+                                metricPathNewUnit,
+                                self.derivative(
+                                                metricPathNewUnit,
+                                                metricValueNewUnit,
+                                                COUNTER_MAX_64BIT
+                                                ),
+                                raw_value=None,
+                                timestamp=timestamp
+                                )
+                        self.publish_metric(metric)
+                    except AttributeError:
+                        continue
```

Comment:
No logging? Exceptions really probably should *not* pass completely silently unless you have an amazingly good reason for explicitly silencing this.